Vanilla Framework lists styles implementation for rST and Markdown

geoffreynyaga opened the pull request:
This PR introduces the `apply_list_classes` function, which enhances the generated HTML by adding custom CSS classes to list elements  ensuring consistency with the Vanilla design system for both unordered and ordered lists.

Support for definition lists has also been introduced.

## Open Issues
The definition lists are correctly rendered in the rst pages, but fail to do so in the Markdown pages. This is without any changes to Ulwazi. For context, rst pages correctly generate the correct html structure for definition lists ( `dl > dt > dd  > p` ) but the markdown pages only produces a `<p>` tag

NOTE: this has been solved in the latest commit. The definition lists needed additional configuration by updating this on conf.py

```diff
myst_enable_extensions = {
    "colon_fence",
+   "deflist"
}
```

geoffreynyaga committed:
lists implementation

geoffreynyaga committed:
lists implementation

geoffreynyaga committed:
lists implementation

geoffreynyaga committed:
lists implementation

izmalk committed:
Add a test suit (#23)

Co-authored-by: nbellol <nbellolugo97@gmail.com>

geoffreynyaga committed:
lists implementation with definition list support for myst

geoffreynyaga committed:
lists implementation with definition list support for myst

izmalk committed:
Add testing workflow

izmalk committed:
Add environment setup steps to workflow

izmalk committed:
Fix too shallow git clone error

izmalk committed:
Add tox install to the tox testing (#25)

Fix testing workflow by adding tox install step to the test command in Makefile.

izmalk commented on docs/content/test_lists_md.md:
```diff
@@ -0,0 +1,86 @@
+
+# Lists (markdown)
```
Suggestion: Delete redundant empty line at the beginning and capitalise the Markdown as the proper name
```suggestion
# Lists (Markdown)
```

izmalk commented on docs/content/test_lists.rst:
```diff
@@ -0,0 +1,101 @@
+          #. Sub-step 2
+
+
+
```
```suggestion
```

izmalk commented on docs/content/test_lists.rst:
```diff
@@ -0,0 +1,101 @@
+
+You can also nest lists:
+
+.. tab-set::
```
Note: This line produces an error when I try to build it. The docs builds, but lists are not rendering correctly.

```
/home/izmalk/test/sphinx/theme/ulwazi/docs/content/test_lists.rst:36: ERROR: Unknown directive type "tab-set".
```

geoffreynyaga replied:
Thank you for noticing this and pointingit out. After rebasing, I enabled the [Sphinx Tabs extension](https://sphinx-tabs.readthedocs.io/en/latest/) to solve the build error.

izmalk commented on ulwazi/__init__.py:
```diff
@@ -104,6 +104,30 @@ def apply_heading_classes(body_html: str) -> str:
+        "li": "p-list__item",
+        "ul.simple": "p-list--unordered p-list--simple",
+        "ol.simple": "p-list--ordered p-list--simple",
+    
```
```suggestion
```

izmalk reviewed (changes requested):
Looks good, but I see an error with the RST lists.
Also, can you please try rebasing it to the latest so we can run the test suit to check for regression? I'm just curious. Don't try too hard to rebase if it creates too many problems with merge conflicts.

geoffreynyaga committed:
rebase and configure sphinx tabs

a-velasco committed:
Simplify HTML for inline code (#29)

* simplify HTML for inline code

* fully delete class element

* remove unused js library

* remove unused library

* remove whitespace

AshleyCliff committed:
feat: adds skip links (#26)

izmalk committed:
Implement Vanilla-based Product menu (#31)

Implement product menu
Make it optional, but enabled by default (for testing)
Add product-menu target to the Makefile
Add product-menu help listing
Delete generic secondary docs targets from help listing

geoffreynyaga committed:
sphinx_tabs

geoffreynyaga committed:
fix tabs syntax to match sphinx_tabs

geoffreynyaga committed:
move the examples to cheat sheet

izmalk committed:
Implement Vanilla Horizontal menu (#32)

Refactored JS code, removing some old, unused fragments and modifying the recommended code snippets from Vanilla Framework to make them work correctly. All automatic and manual tests are working great.
Additional changes are implemented to make sure compatibility with Product menu.

izmalk commented:
1. There are a few conflicts that need to be resolved. I advise to make a rebase or merge the main branch into it.
2. The new files are deleted, but the content was not added to any of the cheat sheets. Is it no longer needed?

msuchane committed:
Task list style (#27)

* Task list style

* Apply Vladimir's suggestions

* Remove reference to a deleted file

a-velasco committed:
Enable Pygments for syntax highlighting (#30)

* enable pygments highlighting in conf.py

* Improving cheatsheets sample docs

Added YAML and Python code blocks to cheat sheets for manual testing

* Add shell syntax highlighting check to cheatsheet

---------

Co-authored-by: Vladimir <48120135+izmalk@users.noreply.github.com>

geoffreynyaga committed:
Merge branch 'main' into lists

geoffreynyaga committed:
[fix] remove unnecessary test files

geoffreynyaga committed:
Merge branch 'main' into lists

geoffreynyaga committed:
[fix] remove duplicate function call

geoffreynyaga committed:
[update] new compiled css

geoffreynyaga commented:
@izmalk I have removed the unnecessary test files since the content/examples are now in the cheatsheet.

nbellol committed:
rebase changes

nbellol reviewed (approved):
LGTM! all works and is now rebased

izmalk committed:
Merge branch 'main' into lists
